lib/palindrome_check.rb

```diff
@@ -1,5 +1,72 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
+
+###all spaces must be ignored!!!
+
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  # raise NotImplementedError
+  return false if my_phrase == nil
+  return true if my_phrase.length == 1
+
+  i = 0
+  j = my_phrase.length - 1
+
+  # puts palindrome_check("han nah  ")
+
+  while i < j do
+    if my_phrase[i] == my_phrase[j]
+      i += 1
+
+      until my_phrase[i] != " " do
+        i += 1
+      end
+
+      j -= 1
+
+      until my_phrase[j] != " " do
```

Comment:
Also continue to check that `i < j` before comparing the character at index j with a white space.